aten/src/ATen/native/mkldnn/BinaryOps.cpp

```diff
@@ -0,0 +1,70 @@
+#include <ATen/ATen.h>
+#include <ATen/Config.h>
+#include <ATen/NativeFunctions.h>
+
+#if !AT_MKLDNN_ENABLED()
+
+namespace at {
+namespace native {
+
+Tensor& mkldnn_add_out(
+    Tensor& result,
+    const Tensor& self,
+    const Tensor& other,
+    Scalar alpha) {
+  AT_ERROR("mkldnn_add_out: ATen not compiled with MKLDNN support");
+}
+
+Tensor mkldnn_add(const Tensor& self, const Tensor& other, Scalar alpha) {
+  AT_ERROR("mkldnn_add: ATen not compiled with MKLDNN support");
+}
+
+Tensor& mkldnn_add_(Tensor& self, const Tensor& other, Scalar alpha) {
+  AT_ERROR("mkldnn_add_: ATen not compiled with MKLDNN support");
+}
+
+} // namespace native
+} // namespace at
+
+#else // AT_MKLDNN_EBABLED
+
+#include <ATen/native/mkldnn/MKLDNNCommon.h>
+
+namespace at {
+namespace native {
+
+Tensor& mkldnn_add_out(
+    Tensor& result,
+    const Tensor& self,
+    const Tensor& other,
+    Scalar alpha) {
+  ideep::tensor& x = itensor_from_mkldnn(self);
+  ideep::tensor& y = itensor_from_mkldnn(other);
+
+  ideep::tensor& z = itensor_from_mkldnn(result);
+  const std::vector<float> scales{1.0, alpha.to<float>()};
```

Comment:
do we ever worry about mkldnn supporting double?  We could technically write this generically with a dispatch macro.

Reply:
[ideep::sum::compute](https://bddppq.github.io/codebrowser/pytorch/pytorch/third_party/ideep/include/ideep/computations.hpp.html#_ZN5ideep3sum7computeERKSt6vectorIfSaIfEERKS1_INS_6tensorESaIS6_EERS6_) only takes [scales](https://bddppq.github.io/codebrowser/pytorch/pytorch/third_party/ideep/include/ideep/abstract_types.hpp.html#ideep::scale_t) as float vector